### PR TITLE
Update the documentation related with the Delegated Identity API

### PR DESCRIPTION
**Pull request by amartinezfayo**
Small documentation changes to reflect that the Delegated Identity API supports both X509-SVIDs and JWT-SVIDs.
It also updates the text to consider the Windows platform.

### CODE REVIEW

File: doc/spire_agent.md [hunk 1]
```diff
@@ -282,9 +282,9 @@ plugins {
 ```
 ## Delegated Identity API
 
-SPIRE agent has support for Delegated Identity API. This API is intended for use cases where a (authorized) workload wants access to the X509-SVIDs and bundles on behalf of another workload. A list of authorized delegates SPIFFE IDs of workloads are defined for this purpose. The API is served over the same endpoint address as the admin API. Based on workload's selectors, a (authorized) workload could be subscribed to get X509-SVIDs and federated bundles from the registered entries that match the provided selectors.
+SPIRE Agent has support for Delegated Identity API. This API is intended for use cases where a (authorized) workload wants access to the SVIDs and bundles on behalf of another workload. A list of authorized delegates SPIFFE IDs of workloads are defined for this purpose. The API is served over the same endpoint address as the admin API. Based on workload's selectors, a (authorized) workload could be subscribed to get SVIDs and federated bundles from the registered entries that match the provided selectors. Both X509-SVIDs and JWT-SVIDs are supported.
```
Comment (azdagron) [inline doc/spire_agent.md:285]: This whole paragraph could actually use a little bit of love.... how about:

```suggestion
The Delegated Identity API allows an authorized (i.e. delegated) workload to obtain SVIDs and bundles on behalf of workloads that cannot be attested by SPIRE Agent directly. The authorized workload does so by providing SPIRE Agent the selectors that would normally be obtained during workload attestation. The Delegated Identity API is served over the admin API endpoint.
```

File: doc/spire_agent.md [hunk 2]
```diff
@@ -282,9 +282,9 @@ plugins {
...
 
-In order to use this API, you shall configure the `admin_socket_path` and `authorized_delegates` (SPIFFE ID list of authorized delegates identities), as the following example:
+In order to use this API, you shall configure the admin API endpoint address and the SPIFFE ID list of authorized delegates identities, as the following example:
```
Comment (azdagron) [inline doc/spire_agent.md:287]: How about:

``` suggestion
To enable the Delegated Identity API, configure the admin API endpoint address and the list of SPIFFE IDs for authorized delegates. For example:
```

**Review by azdagron — COMMENTED** [review]
Thanks for catching this, @amartinezfayo ! I added some suggestions for some changes that are probably a little more broadly scoped than you intended for this PR :)

File: doc/spire_agent.md (hunk 1 above)
Comment (amartinezfayo) [reply doc/spire_agent.md:285]: Much better!

**Pull request merged**